src/layout/mod.rs

```diff
@@ -1335,6 +1335,39 @@ impl<W: LayoutElement> Layout<W> {
         }
     }
 
+    pub fn move_workspace_to_output(&mut self, output: &Output) {
+        let MonitorSet::Normal {
+            monitors,
+            active_monitor_idx,
+            ..
+        } = &mut self.monitor_set
+        else {
+            return;
+        };
+
+        let current = &mut monitors[*active_monitor_idx];
+        if current.active_workspace_idx == current.workspaces.len() - 1 {
+            // Insert a new empty workspace.
+            let ws = Workspace::new(current.output.clone(), current.options.clone());
+            current.workspaces.push(ws);
+        }
+        let mut ws = current.workspaces.remove(current.active_workspace_idx);
+        current.active_workspace_idx = current.active_workspace_idx.saturating_sub(1);
```

Comment:
I suspect that we want a `current.workspace_switch = None;` here followed by a `current.clean_up_workspaces();`. In case we hit that action during a workspace switch.

Reply:
Added, but I think the `clean_up_workspaces` is superfluous: whole workspaces are moved, so at least for `current`, it's not possible that a new empty workspace will exist. At least not without some other window/column moving actions being executed concurrently, and I don't think that's possible?

Reply:
It's possible:
1. Start a workspace switch on one monitor
2. Some window closes on the same monitor, maybe even on a different workspace entirely. Since a workspace switch is ongoing, the workspaces aren't cleaned up
3. The user moves a workspace, either from this monitor, or to this monitor
4. We cancel the workspace switch here
5. If we don't clean up workspaces right away, that workspace from step 2 remains not cleaned up

Reply:
Thanks for the explanation! I didn't connect that the cleanup happens when the animation is done.